### ISSUE TITLE
Single image inversion 

### ISSUE THREAD

**Issue by Dong142857** [Twizwei/in-n-out#4]
Can you provide a single image inversion guidance? I use my own dataset to train but this training code may need a latent.pt as initialization. How to get the lantent code?

**Comment by boringwar** [comment]
It seems that the indomain inversion is kept secret by the author.

I used a third-party w+ invesion, and the dimension is 18 instead of 14.

**Comment by boringwar** [comment]
why the author does not claim clearly that custom dataset will fail to run in run_train.sh if without "latents.pt"

**Comment by Dong142857** [comment]
The W+ dimension in EG3D is exactly 14, not 18, because the feature map upsamples to 256x256 instead of 1024x1024. To obtain the latent code, i think we can try and modify this [code](https://github.com/Twizwei/in-n-out/blob/main/eg3d/inversion/invert_multi_w.py), or use the pretrained EG3D inversion encoder such as [GOAE](https://github.com/jiangyzy/GOAE) and [TriplaneNet](https://github.com/anantarb/triplanenet). I think the author has opened lots of their codes, but there is too many codes to read without guidance.

**Comment by Twizwei** [comment]
Yep I can provide a script for single image inversion this month. Sorry a little bit occupied by other stuff.

**Comment by Twizwei** [comment]
> The W+ dimension in EG3D is exactly 14, not 18, because the feature map upsamples to 256x256 instead of 1024x1024. To obtain the latent code, i think we can try and modify this [code](https://github.com/Twizwei/in-n-out/blob/main/eg3d/inversion/invert_multi_w.py), or use the pretrained EG3D inversion encoder such as [GOAE](https://github.com/jiangyzy/GOAE) and [TriplaneNet](https://github.com/anantarb/triplanenet). I think the author has opened lots of their codes, but there is too many codes to read without guidance.

Yes you are right. For EG3D its latent space is 14-dim, instead of 18-dim like 2D StyleGANs. I'll try to find a slot recently to provide you with a script.
